### PR TITLE
Libraries and locations

### PR DESCRIPTION
**Pull request by eliotjordan**
Adds library and location classes.

### CODE REVIEW

File: lib/alma/api_defaults.rb [hunk 1]
```diff
@@ -27,6 +27,10 @@ def items_base_path
       "#{self.region}/almaws/v1/items"
     end
 
+    def configuraion_base_path
```
Comment (sensei100) [inline lib/alma/api_defaults.rb]:
```suggestion
    def configuration_base_path
```

File: lib/alma/library.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+module Alma
+  class Library < AlmaRecord
+    extend Alma::ApiDefaults
+
+    def self.all(args: {})
+      response = HTTParty.get("#{configuraion_base_path}/libraries", query: args, headers: headers, timeout: timeout)
```
Comment (sensei100) [inline lib/alma/library.rb]:
```suggestion
      response = HTTParty.get("#{configuration_base_path}/libraries", query: args, headers: headers, timeout: timeout)
```

File: lib/alma/library.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      if response.code == 200
+        LibrarySet.new(response)
+      else
+        raise StandardError, get_body_from(response)
+      end
+    end
+
+    def self.find(library_code:, args: {})
+      response = HTTParty.get("#{configuraion_base_path}/libraries/#{library_code}",query: args, headers: headers, timeout: timeout)
```
Comment (sensei100) [inline lib/alma/library.rb]:
```suggestion
      response = HTTParty.get("#{configuration_base_path}/libraries/#{library_code}",query: args, headers: headers, timeout: timeout)
```

File: lib/alma/location.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
+module Alma
+  class Location < AlmaRecord
+    extend Alma::ApiDefaults
+
+    def self.all(library_code:, args: {})
+      response = HTTParty.get("#{configuraion_base_path}/libraries/#{library_code}/locations", query: args, headers: headers, timeout: timeout)
```
Comment (sensei100) [inline lib/alma/location.rb]:
```suggestion
      response = HTTParty.get("#{configuration_base_path}/libraries/#{library_code}/locations", query: args, headers: headers, timeout: timeout)
```

File: lib/alma/location.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+      if response.code == 200
+        LocationSet.new(response)
+      else
+        raise StandardError, get_body_from(response)
+      end
+    end
+
+    def self.find(library_code:, location_code:, args: {})
+      response = HTTParty.get("#{configuraion_base_path}/libraries/#{library_code}/locations/#{location_code}", query: args, headers: headers, timeout: timeout)
```
Comment (sensei100) [inline lib/alma/location.rb]:
```suggestion
      response = HTTParty.get("#{configuration_base_path}/libraries/#{library_code}/locations/#{location_code}", query: args, headers: headers, timeout: timeout)
```

**Review by sensei100 — CHANGES_REQUESTED** [review]
@eliotjordan Thanks for adding this to the project.  I suggested fixing the typo in the base_path in a number of places.

File: lib/alma/api_defaults.rb (hunk 1 above)
Comment (eliotjordan) [reply lib/alma/api_defaults.rb]: Fixed and rebased.

**Comment by eliotjordan** [comment]
@sensei100 Thanks for the review and corrections! I made the updates you suggested and rebased to keep the git history clean.

**Review by sensei100 — APPROVED** [review]

**Pull request merged**